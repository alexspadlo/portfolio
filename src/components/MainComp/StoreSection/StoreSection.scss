@import "../../../stylesheet/_vars.scss";

.boxStyle {
  border: 1px solid #caa24f;
}

.minHeight {
  min-height: 531px;
}

.centerAlign {
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
}

.hidden {
  opacity: 1;
  animation: first 3s, second 4s, third 3s;
}

@keyframes first {
  from {
    position: absolute;
    bottom: -100vh;
  }

  to {
    bottom: 0;
  }
}

@keyframes second {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}

@keyframes third {
  from {
    position: absolute;
  }

  to {
    position: relative;
  }
}
