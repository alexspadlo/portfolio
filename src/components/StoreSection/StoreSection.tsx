import React, { FC } from 'react'
import shopPaint from '../../static/images/forShop.png'
import s from './StoreSection.scss'

const StoreSection: FC = () => {
    return (
        <div className='container-fluid my-5 py-4'>
            <div className='row py-5'>
                <div className='col-12 col-lg-7 m-auto'>
                    <section className={s.boxStyle}>
                        <div className='row m-0'>
                            <div className='col-12 col-lg-6 p-0'>
                                <img className="img-fluid w-100" src={shopPaint} alt={shopPaint} />
                            </div>
                            <div className='col-12 col-lg-6 p-0'>
                                <section className='w-100 text-center py-5 py-lg-4 px-4 h-100 position-relative'>
                                    <section className={`${s.centerAlign} position-relative position-lg-absolute`}>
                                        <p className={`${s.fontBold} h4`}>Support my Art</p>
                                        <p className={`${s.fontThin}`}>Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>
                                        <span className="btn mt-3 btn-info">View shop</span>
                                    </section>
                                </section>
                            </div>
                        </div>
                    </section>
                </div>
            </div>
        </div>

    )
}

export default StoreSection