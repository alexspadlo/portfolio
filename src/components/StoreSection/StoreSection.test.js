import React from 'react'
import renderer from 'react-test-renderer'
import StoreSection from './StoreSection'

describe('Test on <StoreSection /> component', () => {
    test('Should render without the props', () => {
        const componentNoprops = renderer.create(<StoreSection />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
})
