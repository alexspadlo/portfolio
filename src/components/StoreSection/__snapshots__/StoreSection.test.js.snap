// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <StoreSection /> component Should render without the props 1`] = `
<div
  className="container-fluid my-5 py-4"
>
  <div
    className="row py-5"
  >
    <div
      className="col-12 col-lg-7 m-auto"
    >
      <section
        className="boxStyle"
      >
        <div
          className="row m-0"
        >
          <div
            className="col-12 col-lg-6 p-0"
          >
            <img
              alt="test-file-stub"
              className="img-fluid w-100"
              src="test-file-stub"
            />
          </div>
          <div
            className="col-12 col-lg-6 p-0"
          >
            <section
              className="w-100 text-center py-5 py-lg-4 px-4 h-100 position-relative"
            >
              <section
                className="centerAlign position-relative position-lg-absolute"
              >
                <p
                  className="fontBold h4"
                >
                  Support my Art
                </p>
                <p
                  className="fontThin"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </p>
                <span
                  className="btn mt-3 btn-info"
                >
                  View shop
                </span>
              </section>
            </section>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
